service/evidence/evidence_store.go

```diff
@@ -36,63 +36,66 @@ func init() {
 }
 
 // StoreEvidence is a method implementation of the evidenceServer interface: It receives an req and stores it
-func (s *Service) StoreEvidence(_ context.Context, req *evidence.StoreEvidenceRequest) (*evidence.StoreEvidenceResponse, error) {
-	var (
-		resp = &evidence.StoreEvidenceResponse{}
-		err  error
-		e    = req.GetEvidence()
-	)
+func (s *Service) StoreEvidence(_ context.Context, req *evidence.StoreEvidenceRequest) (resp *evidence.StoreEvidenceResponse, err error) {
 
-	_, err = e.Validate()
-	if err != nil {
-		log.Errorf("Invalid evidence: %v", err)
-		newError := fmt.Errorf("invalid evidence: %w", err)
+	err = s.handleEvidence(req.GetEvidence())
 
-		// Inform our hook, if we have any
-		if s.EvidenceHook != nil {
-			for _, hook := range s.EvidenceHook {
-				go hook(nil, newError)
-			}
+	if err != nil {
+		resp = &evidence.StoreEvidenceResponse{
+			Status: false,
 		}
 
-		return resp, status.Errorf(codes.InvalidArgument, "invalid req: %v", err)
+		return resp, status.Errorf(codes.Internal, "Error while handling evidence: %v", err)
```

Comment:
This is probably the correct PR for this: The problem that I see here is that it is now indistinguishable whether we have an internal error (such as the DB connection was lost or so), which the user cannot fix from an invalid argument error, which the user can fix, e.g. by supplying the correct argument.

Reply:
So I would suggest to at least call the validate function outside of the handleXX function. All errors from validate should be invalid arguments. all errors from the handleXXX can be internal errors.

Reply:
@oxisto I changed it. But I think it it sufficient if the StoreEvidences() calls the StoreEvidence(). We don't need the handle method. Please, can you take a quick look: https://github.com/clouditor/clouditor/blob/8e6f24de89f83fca5b573138a8f24edb41ea5e60/service/evidence/evidence_store.go#L39

Reply:
Should we than do the same for orchestrator.StoreAssessmentResult(s)()? There is the same error regarding the status message Internal vs InvalidArgument.